modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/sqlquery/SqlQueryEnrichment.scala

```diff
@@ -134,18 +142,35 @@ final case class SqlQueryEnrichment[F[_]: Monad: DbExecutor](
     customContexts: List[SelfDescribingData[Json]],
     unstructEvent: Option[SelfDescribingData[Json]]
   ): F[ValidatedNel[FailureDetails.EnrichmentFailure, List[SelfDescribingData[Json]]]] = {
+    def useConnection(connection: Connection): F[Either[NonEmptyList[String], List[SelfDescribingData[Json]]]] =
+      (for {
+        placeholders <- Input
+                          .buildPlaceholderMap(inputs, event, derivedContexts, customContexts, unstructEvent)
+                          .toEitherT[F]
+                          .leftMap { t =>
+                            NonEmptyList.of("Error while building the map of placeholders", t.toString)
+                          }
+        verifiedPlaceholders <-
+          EitherT(blocker.blockOn(Monad[F].pure(DbExecutor.allPlaceholdersFilled(connection, query.sql, placeholders))))
+            .leftMap { t =>
+              NonEmptyList.of("Error while filling the placeholders", t.toString)
+            }
+        result <- verifiedPlaceholders match {
```

Comment:
These steps in the forcomp are getting bigger, maybe they could be extracted to separate methods? I think it would be easier to follow what `useConnection` does :)